pyoptsparse/pySNOPT/pySNOPT.py

```diff
@@ -484,11 +521,28 @@ def __call__(
 
         else:  # We are not on the root process so go into waiting loop:
             self._waitLoop()
+            cw = iw = rw = xs = hs = pi = None
 
         # Communication solution and return
         commSol = self._communicateSolution(sol)
-
-        return commSol
+        cw = self.optProb.comm.bcast(cw, root=0)
+        iw = self.optProb.comm.bcast(iw, root=0)
+        rw = self.optProb.comm.bcast(rw, root=0)
+        xs = self.optProb.comm.bcast(xs, root=0)
+        hs = self.optProb.comm.bcast(hs, root=0)
+        pi = self.optProb.comm.bcast(pi, root=0)
+        if self.getOption("Return work arrays"):
+            restartDict = {
+                "cw": cw,
+                "iw": iw,
+                "rw": rw,
+                "xs": xs,
+                "hs": hs,
+                "pi": pi,
+            }
+            return commSol, restartDict
```

Comment:
Is having a different number of returned arguments depending on an option the best practice?
We could work around this by having another couple of methods called `setRestartDict` and `getRestartDict` to avoid this user-facing API change

Reply:
I personally think this is fine, since the user has control over this by specifying a Python option. However, I'm open to changing the API to make it more user-friendly.

Instead of having a separate function, what about specifying the file path where pySNOPT would write the dictionary as a file (either pickle or numpy format, we can decide). Then, we can have `__call__(..., restartFile=<file path>)` accept the file path. That way, saving/loading the restart file would be handled by pyOptSparse instead of the user.

Reply:
I am fine with the current return approach. Are you using file I/O for your cases @nwu63 or passing the dictionary through memory? I can see the file approach being better if the use case requires the user to do it anyway.

Reply:
My idea of adding the additional methods was just to have a "lighter" modification than the internal I/O, but if you are willing to do so I think that the `<file-path>` approach would be the smoothest approach on the user side! Not sure what data format is best, in my mind 1) we should not add more dependencies (e.g. json?) 2) (optional) it would be nice for the database to be human readable or easily accessible in case we want to double check what's in there

Reply:
I don't think human-readable is useful, these are just a bunch of fairly long numpy arrays that are not human-readable anyways.

Reply:
alright, let's ignore that then!

Reply:
We agreed on not changing this for now, right?

Reply:
Yeah, I think we should keep this as is for now, and if we want to change the API we can do that later.